game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/sub/OffensiveSubsRetreatTest.java

```diff
@@ -0,0 +1,196 @@
+    final OffensiveSubsRetreat offensiveSubsRetreat =
+        new OffensiveSubsRetreat(battleState, battleActions);
+
+    offensiveSubsRetreat.execute(executionStack, delegateBridge);
+
+    verify(battleActions, never()).queryRetreat(anyBoolean(), any(), any(), any());
+  }
+
+  static List<Arguments> retreatDoesNotHappen() {
+    return List.of(
+        Arguments.of(
+            "Battle is over",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .over(true)
+                .build()),
+        Arguments.of(
+            "Defending Destroyer exists",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .defendingUnits(List.of(givenUnitDestroyer()))
+                .build()),
+        Arguments.of(
+            "Waiting to Die Defending Destroyer exists",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .defendingWaitingToDie(List.of(givenUnitDestroyer()))
+                .build()),
+        Arguments.of(
+            "Amphibious assault",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData().withTransportCasualtiesRestricted(false).build())
+                .amphibious(true)
+                .build()),
+        Arguments.of(
+            "Can withdraw but only defenseless transports on the defense",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData()
+                        .withTransportCasualtiesRestricted(true)
+                        .withSubmersibleSubs(false)
+                        .build())
+                .defendingUnits(List.of(givenUnitTransport()))
+                .attackerRetreatTerritories(List.of(mock(Territory.class)))
+                .build()),
+        Arguments.of(
+            "Can submerge but only defenseless transports on the defense",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData()
+                        .withTransportCasualtiesRestricted(true)
+                        .withSubmersibleSubs(true)
+                        .build())
+                .defendingUnits(List.of(givenUnitTransport()))
+                .build()),
+        Arguments.of(
+            "No retreat territories and can not submerge",
+            givenBattleStateBuilder()
+                .gameData(
+                    MockGameData.givenGameData().withTransportCasualtiesRestricted(false).build())
+                .attackerRetreatTerritories(List.of())
+                .build()));
+  }
+
+  static class MockGameData {
```

Comment:
What do you think of this class?  I was wanting a fluent way to generate a mock GameData.